packages/rrweb/src/replay/index.ts

```diff
@@ -2113,24 +2192,33 @@ export class Replayer {
         return;
       }
       const ctx = this.mouseTail.getContext('2d');
-      if (!ctx || !this.tailPositions.length) {
+      if (!ctx || !this.pointers[pointerId].tailPositions.length) {
         return;
       }
       ctx.clearRect(0, 0, this.mouseTail.width, this.mouseTail.height);
       ctx.beginPath();
       ctx.lineWidth = lineWidth;
       ctx.lineCap = lineCap;
       ctx.strokeStyle = strokeStyle;
-      ctx.moveTo(this.tailPositions[0].x, this.tailPositions[0].y);
-      this.tailPositions.forEach((p) => ctx.lineTo(p.x, p.y));
+      ctx.moveTo(
+        this.pointers[pointerId].tailPositions[0].x,
+        this.pointers[pointerId].tailPositions[0].y,
+      );
+      this.pointers[pointerId].tailPositions.forEach((p) =>
+        ctx.lineTo(p.x, p.y),
+      );
       ctx.stroke();
     };
 
-    this.tailPositions.push(position);
+    this.pointers[pointerId].tailPositions.push(position);
     draw();
     setTimeout(() => {
-      this.tailPositions = this.tailPositions.filter((p) => p !== position);
-      draw();
+      if (pointerId in this.pointers) {
```

Comment:
this `if` statement fixes the issue that was causing null access to the deleted pointer

Reply:
An alternative to `delete` is to assign `null` or `undefined` to `this.pointers[id]`. This accomplishes what we need: eliminate the reference to the object that is at `this.pointers[id]` so that it's eligible for garbage collection.

[`delete` is also about 2x slower than re-assigning](https://jsperf.app/delete-vs-undefined-vs-null/6). When you would use `delete` is if you actually need to iterate through the entries of `this.pointers` (which it doesn't seem like we do).

Reply:
i tried this, and then i got about 70 errors complaining about potentially unsafe references to a value that could be `null` or `undefined` 😭 so i'll need to add a bunch of `?`

Reply:
Ahh yeah, if [noUncheckedIndexAccess](https://www.typescriptlang.org/tsconfig/#noUncheckedIndexedAccess) were turned on we would get these errors when calling `delete` as well. Since `pointers` is defined as `Record<string, {...}>`, it interprets any key access as `{...}` type. 

This code block is probably the most susceptible to this because it is called in a `setTimeout`, so code here will run in a future event loop. e.g. you could have a scenario like:

* draw mouse tail
* setTimeout called
* touchEnd happens (delete pointer)
* setTimeout fn runs


My suggestion, let's merge this PR and get multi touch working on sentry to unblock our SDK engineers. Then you can come back to change this to be more safe when we prepare to push this upstream